bob/src/hw_metrics_collector.rs

```diff
@@ -179,52 +184,71 @@ enum CommandError {
     Primary(String),
 }
 struct CPUStatCollector {
-    iostat_avl: bool,
+    procfs_avl: bool,
 }
 
 impl CPUStatCollector {
     fn new() -> CPUStatCollector {
-        CPUStatCollector { iostat_avl: true }
+        CPUStatCollector {
+            procfs_avl: true
+        }
     }
 
-    fn parse_iostat_result(iostat_str: &str) -> Result<f64, std::num::ParseFloatError> {
-        iostat_str.replace(',', ".").parse()
+    fn stat_cpu_line() -> Result<String, String> {
+        let lines = file_contents(CPU_STAT_FILE)?;
+        for stat_line in lines {
+            let mut parts = stat_line.split_whitespace();
+            if let Some("cpu") = parts.next() {
```

Comment:
It is better to use `starts_with` to find the line and only after the line is found, perform string splitting